app/pods/publications/publication/documents/controller.js

```diff
@@ -57,4 +94,132 @@ export default class PublicationDocumentsController extends Controller {
   toggleFolderCollapse(piece) {
     piece.set('collapsed', !piece.collapsed);
   }
+
+  @action
+  uploadPiece(file) {
+    const now = moment().utc()
+      .toDate();
+    const documentContainer = this.store.createRecord('document-container', {
+      created: now,
+    });
+    const piece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      file: file,
+      accessLevel: this.defaultAccessLevel,
+      confidential: false,
+      name: file.filenameWithoutExtension,
+      documentContainer: documentContainer,
+    });
+    this.newPieces.pushObject(piece);
+  }
+
+  @task
+  *savePieces() {
+    const savePromises = this.newPieces.map(async(piece) => {
+      try {
+        this.savePiece.perform(piece);
+      } catch (error) {
+        this.deletePiece.perform(piece);
+        throw error;
+      }
+    });
+    yield all(savePromises);
+    this.isOpenPieceUploadModal = false;
+    this.newPieces = A();
+  }
+
+  /**
+   * Save a new document container and the piece it wraps
+   */
+  @task
+  *savePiece(piece) {
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.save();
+    yield piece.save();
+    const pieces = yield this.model.case.hasMany('pieces').reload();
+    pieces.pushObject(piece);
+    yield this.model.case.save();
+  }
+
+  @task
+  *cancelUploadPieces() {
+    const deletePromises = this.newPieces.map((piece) => this.deletePiece.perform(piece));
+    yield all(deletePromises);
+    this.newPieces = A();
+    this.isOpenPieceUploadModal = false;
+  }
+
+  @task
+  *deletePiece(piece) {
+    const file = yield piece.file;
+    yield file.destroyRecord();
+    this.newPieces.removeObject(piece);
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.destroyRecord();
+    yield piece.destroyRecord();
+  }
+
+
+  /** TRANSLATION ACTIVITIES **/
+
+  @action
+  openTranslationRequestModal() {
+    this.translateActivity.finalTranslationDate = this.model.publicationFlow.translateBefore;
+    this.translateActivity.pieces = this.selectedPieces;
+    this.showTranslationModal = true;
+  }
+
+  get getTranslateActivityBeforeDate() {
+    console.log(this.model.publicationFlow.translateBefore);
+    if (this.model.publicationFlow.translateBefore) {
+      return this.model.publicationFlow.translateBefore;
+    }
+    return new Date();
+  }
+
+  @action
+  async saveTranslationActivity() {
+    this.showLoader = true;
+    this.showTranslationModal = false;
+    const translateSubCaseType = EmberObject.create({
+      id: CONFIG.SUBCASE_TYPES.vertalen.id,
+      uri: CONFIG.SUBCASE_TYPES.vertalen.url,
+    });
+
+    // TODO take from other subcase maybe?
+    const shortTitle = await this.model.case.shortTitle;
+    const title = await this.model.case.title;
```

Comment:
longTitle van case kan undefined zijn.
Dossiers aangemaakt via MR hebben normaal geen lange titel. Die optie is er niet.
longTitle via MR moet dan sowieso van bestaande subcases komen